nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/classloader/ClassLoaderUtils.java

```diff
@@ -143,19 +143,13 @@ public static String generateAdditionalUrlsFingerprint(Set<URL> urls) {
 
         //Sorting so that the order is maintained for generating the fingerprint
         Collections.sort(listOfUrls);
-        try {
-            MessageDigest md = MessageDigest.getInstance("MD5");
-            listOfUrls.forEach(url -> {
-                urlBuffer.append(url).append("-").append(getLastModified(url)).append(";");
-            });
-            byte[] bytesOfAdditionalUrls = urlBuffer.toString().getBytes(StandardCharsets.UTF_8);
-            byte[] bytesOfDigest = md.digest(bytesOfAdditionalUrls);
-
-            return DatatypeConverter.printHexBinary(bytesOfDigest);
-        } catch (NoSuchAlgorithmException e) {
-            LOGGER.error("Unable to generate fingerprint for the provided additional resources {}", new Object[]{urls, e});
-            return null;
-        }
+        listOfUrls.forEach(url -> {
```

Comment:
Hi,
Optinally, this statement clambda an be replaced with an expression lambda. Also, do you think changing urlBuffer type from StringBuffer to StringBuilder as it doesn't need to be synchronised, would have any performance improvement effect?

Reply:
The `StringBuffer` declaration was not part of this initial change, but there doesn't appear to be any need for it, so will replace with `StringBuilder` and use an expression lambda.